code/modules/mob/living/carbon/human/species/kidan.dm

```diff
@@ -43,7 +43,162 @@
 		"is stabbing themselves with their mandibles!",
 		"is holding their breath!")
 
-
 /datum/species/kidan/get_species_runechat_color(mob/living/carbon/human/H)
 	var/obj/item/organ/internal/eyes/E = H.get_int_organ(/obj/item/organ/internal/eyes)
 	return E.eye_color
+
+// They receive a HUD for seeing and an action to create pheromones
+/datum/species/kidan/on_species_gain(mob/living/carbon/human/H)
+	..()
+	// HUD for detecting pheromones
+	var/datum/atom_hud/kidan_hud = GLOB.huds[DATA_HUD_KIDAN_PHEROMONES]
+	kidan_hud.add_hud_to(H)
+
+	// Action for creating pheromones
+	var/datum/action/innate/produce_pheromones/produce_pheromones = new()
+	produce_pheromones.Grant(H)
+
+/datum/species/kidan/on_species_loss(mob/living/carbon/human/H)
+	..()
+	// Removing the HUD for detecting pheromones
+	var/datum/atom_hud/kidan_hud = GLOB.huds[DATA_HUD_KIDAN_PHEROMONES]
+	kidan_hud.remove_hud_from(H)
+
+	// Removing the action for creating pheromones
+	for(var/datum/action/innate/produce_pheromones/action in H.actions)
+		action.Remove(H)
+
+// Pheromones spawnable by kida, only perceivable by other kida
```

Comment:
```suggestion
/// Pheromones spawnable by kida, only perceivable by other kida
```

Reply:
I did not know atoms can be autodocced too, good to know